gnovm/pkg/packages/imports.go

```diff
@@ -50,44 +49,60 @@ func Imports(pkg *std.MemPackage, fset *token.FileSet) (ImportsMap, error) {
 	return res, nil
 }
 
-// FileImport represents an import
+// FileImport represents an import and it's location in the source file
 type FileImport struct {
 	PkgPath string
 	Spec    *ast.ImportSpec
 }
 
 // FileImports returns the list of gno imports in the given file src.
 // The given filename is only used when recording position information.
-func FileImports(filename string, src string, fset *token.FileSet) ([]FileImport, error) {
+func FileImports(filename string, src string, fset *token.FileSet) ([]*FileImport, error) {
 	if fset == nil {
 		fset = token.NewFileSet()
 	}
 	f, err := parser.ParseFile(fset, filename, src, parser.ImportsOnly)
 	if err != nil {
 		return nil, err
 	}
-	res := make([]FileImport, len(f.Imports))
-	for i, im := range f.Imports {
-		importPath, err := strconv.Unquote(im.Path.Value)
+	res := make([]*FileImport, len(f.Imports))
+	for i, spec := range f.Imports {
+		pkgPath, err := strconv.Unquote(spec.Path.Value)
 		if err != nil {
 			// should not happen - parser.ParseFile should already ensure we get
 			// a valid string literal here.
-			panic(fmt.Errorf("%v: unexpected invalid import path: %v", fset.Position(im.Pos()).String(), im.Path.Value))
+			panic(fmt.Errorf("%v: unexpected invalid import path: %v", fset.Position(spec.Pos()).String(), spec.Path.Value))
 		}
 
-		res[i] = FileImport{
-			PkgPath: importPath,
-			Spec:    im,
+		res[i] = &FileImport{
+			PkgPath: pkgPath,
+			Spec:    spec,
 		}
 	}
 	return res, nil
 }
 
-type ImportsMap map[FileKind][]FileImport
+type ImportsMap map[FileKind][]*FileImport
+
+func (imap ImportsMap) ToStrings() map[FileKind][]string {
+	res := make(map[FileKind][]string, len(imap))
+	for k, v := range imap {
+		c := make([]string, 0, len(v))
+		for _, x := range v {
+			c = append(c, x.PkgPath)
+		}
```

Comment:
should we check x != nil first ? could be unnecessary if it cannot be nil

Reply:
it can only be nil if the user inserted nil itself by mistake, in this case I think the panic is more appropriate to catch the problem early rather than silently ignoring the value